src/main/scala/inox/utils/RustInteropGenerator.scala

```diff
@@ -0,0 +1,404 @@
+/* Copyright 2009-2018 EPFL, Lausanne */
+
+package inox
+package utils
+
+import java.io.{File, FileWriter}
+
+import scala.reflect.runtime.universe._
+import scala.collection.mutable.{ArrayBuffer, HashMap => MutableMap, HashSet => MutableSet}
+
+trait RustInteropGeneration { self: InoxSerializer =>
```

Comment:
I hadn't realized this was going to live inside Inox. Is there any point having it here rather than in Stainless? I'd prefer if Inox stayed mostly frontend-independent (if possible).

Reply:
I understand that. We absolutely need this file to generate our AST trees in Rust but as long as it still works, nothing speaks against moving this to Stainless. Do you think everything will still be accessible there?

Reply:
OK, I tested it and there is no problem if we move this entire file to Stainless. See here: https://github.com/epfl-lara/stainless/pull/1060